test/MerkleLib.Fixture.ts

```diff
@@ -0,0 +1,17 @@
+import { getContractFactory } from "./utils";
+import hre from "hardhat";
+
+export const merkleLibFixture = hre.deployments.createFixture(async ({ deployments }) => {
+  await deployments.fixture();
+  const [signer] = await hre.ethers.getSigners();
+  const merkleLib = await (await getContractFactory("MerkleLib", signer)).deploy();
+  const merkleLibTest = await (
+    await hre.ethers.getContractFactory("MerkleLibTest", {
```

Comment:
there is an argument to put this linking deployment within the `getContractFactory` from `utils` so you dont need to create your own ethers factory with `getContractFactory` and then this would be hidden from this fixture. what I describe is more similar to what was done in the hub pool tests. downside with that is the `utils` needs branching logic based on the contract name to do the associated library linking. curious what pattern you think is best and if it warrants changing what's in the `HubPool`'s fixture, [here](https://github.com/across-protocol/across-smart-contracts-v2/blob/a44eeb28839345fab140f80b2de7ad00cfd7968c/test/utils.ts?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L12)

Reply:
Yeah, I think I might've left a comment on that PR (can't remember if I deleted it before sending). I don't love branching on individual contract names.

One compromise between the two approaches is have the second argument mirror the structure of the ethers method, `Signer | Options` where options could contain the _signer_ and the libraries. Then the user can still use the function, but they can pick the library deployment they want to link.

Reply:
So I made the change I mention above. PTAL and let me know what you think.